modules/dictionary/php/fields.class.inc

```diff
@@ -0,0 +1,132 @@
+<?php declare(strict_types=1);
+namespace LORIS\dictionary;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+/**
+ * The dictionary/fields endpoint handles requests to update a specific
+ * field description
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+
+class Fields extends \NDB_Page
+{
+    /**
+     * {@inheritDoc}
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return $user->hasPermission('data_dict_edit');
+    }
+
+    /**
+     * The modules class overrides the default behaviour to handle PUT
+     * requests for the fields endpoint.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $user = $request->getAttribute("user");
+        if ($user === null && !($user instanceof \User)) {
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                "No valid user"
+            );
+        }
+        if (!($this->_hasAccess($user))) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        $path = $request->getURI()->getPath();
+        if (strpos($path, "fields/") === false) {
+            // if the path doesn't contain "fields" we shouldn't have
+            // delegated to this class..
+            throw new \LorisException("Invalid internal LORIS state");
+        }
+        $name = substr($path, strpos($path, "fields/") + strlen("fields/"));
+        switch ($request->getMethod()) {
+        case 'PUT':
+            $status = $this->_replaceDescription(
+                $name,
+                $request->getBody()->__toString()
+            );
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withHeader("X-StatusDesc", $status)
+                ->withStatus(201)
```

Comment:
I would go with 202 since 201 do not fit the situation when an override is deleted.

Reply:
But this is in the case where it's handling a PUT request?

Reply:
Oh, I see. `_replaceDescription` can delete. (It's been a while since I looked at this code..)